### DISCUSSION TITLE
DockerContainer Infrastructure: docker engine host url parameter?

### DISCUSSION THREAD

**Discussion by lezardo76** [PrefectHQ/prefect#8917] [General] [answered]
Hi,

From the documentation, if we want to choose the DockerContainer infrastructure in our deployment, it is mentioned that a docker engine must be present. I assume that the docker engine is local, side by side with the prefect server. How could I change the docker host so that it runs on another docker server? The docker host URL is not present in the DockerContainer block .. but may be there is a trick somewhere by changing some env variables + ssh keys?

Tks,

**Comment by flapili** [comment, marked as answer]
Hi, I believe you can configure the docker daemon to use a specific host as described here : https://docs.docker.com/config/daemon/remote-access/
but I have no clue about how that will work with prefect 😅

**Reply by lezardo76** [reply]
Thanks for this info!

I looked at the code a bit and it looks like the prefect agent expects to have a local docker engine ready if we use DockerContainer infrastructure in a prefect flow deployment. By default, it's doing a docker.from_env() to get the docker client. So, it could be possible to point to another docker server by setting DOCKER_HOST, DOCKER_TLS_VERIFY and DOCKER_CERT_PATH in global environment where the agent is running.

That should do the trick .. to be tested ;)

Tks.
